core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -169,7 +160,7 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def instantiateClass(cls: Class[_], outer: AnyRef, inInterpreter: Boolean): AnyRef = {
     // logInfo("Creating a " + cls + " with outer = " + outer)
-    if (!inInterpreter) {
+    if (inInterpreter) {
```

Comment:
I wouldn't change this unless you can point to a very specific reason you want to.